### PR TITLE
[fix]: [CDS-39968]: Added artifactPath info for active instances and deployments

### PR DESCRIPTION
**Pull request by Abhishek-CDC**
## Describe your changes

- Since we are storing ArtifactSummary in PlanExecutionsSummary table (mongoDB), Made Key sets artifactPathNameSet and tagNameSet for getting actifact_image and tag info in timescale(service_infra_info table) respectively.

- Added artifact_path field to InstanceNg table (mongoDB) to store ArtifactSummary display name.

- Added artifact_path to the response of Active Service Deployments and Active Service Instances.

- removed unused variables infraIdentifier and infraName from InstanceSyncServiceImpl.java

-  removed groupedByEnvironments method with 'For loop' implementation in groupedByArtifacts method.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35893)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 9 Issues 
* Code issue 9 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35893)

**Comment by Abhishek-CDC** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java [hunk 1]
```diff
@@ -1680,53 +1682,60 @@ public InstanceGroupedByArtifactList getInstanceGroupedByArtifactList(
...
+      buildIdToArtifactPathMap.putIfAbsent(buildId, artifactPath);
     });
     List<InstanceGroupedByArtifactList.InstanceGroupedByArtifact> instanceGroupedByArtifactList =
-        groupedByArtifacts(buildEnvInfraMap, envIdToEnvNameMap);
+        groupedByArtifacts(buildEnvInfraMap, envIdToEnvNameMap, buildIdToArtifactPathMap);
 
     return InstanceGroupedByArtifactList.builder().instanceGroupedByArtifactList(instanceGroupedByArtifactList).build();
   }
 
+  private String getArtifactPathFromArtifactPathInfo(String artifactPathInfo) {
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:1693]: In general, as a good coding practise, we should avoid returning null from a method (https://www.codebyamir.com/blog/stop-returning-null-in-java). You can return Optional<String> in this case or an empty string

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java [hunk 2]
```diff
@@ -1680,53 +1682,60 @@ public InstanceGroupedByArtifactList getInstanceGroupedByArtifactList(
...
+    if (artifactPathInfo != null) {
+      String[] res = artifactPathInfo.split(":");
+      int count = res.length;
+      if (count > 1) {
+        return res[0];
+      }
+    }
+    return null;
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:1701]: same comment for returning null as above

File: 126-instance/src/main/java/io/harness/service/deploymentevent/DeploymentEventListener.java [hunk 3]
```diff
@@ -175,14 +175,17 @@ private void setArtifactDetails(Ambiance ambiance, DeploymentSummaryDTO deployme
...
     ArtifactsOutcome artifactsOutcome = (ArtifactsOutcome) optionalOutcome.getOutcome();
-    deploymentSummaryDTO.setArtifactDetails(ArtifactDetails.builder()
-                                                .tag(artifactsOutcome.getPrimary().getTag())
-                                                .artifactId(artifactsOutcome.getPrimary().getIdentifier())
-                                                .build());
+    deploymentSummaryDTO.setArtifactDetails(
+        ArtifactDetails.builder()
+            .tag(artifactsOutcome.getPrimary().getTag())
+            .artifactId(artifactsOutcome.getPrimary().getIdentifier())
+            .artifactPathInfo(artifactsOutcome.getPrimary().getArtifactSummary().getDisplayName())
```
Comment (yogesh-chauhan) [inline 126-instance/src/main/java/io/harness/service/deploymentevent/DeploymentEventListener.java:187]: can we call this field displayName itself ?

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 8 of 11 files at r1, 1 of 1 files at r2, 2 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35893)* status: 1 file(s) to review. 0 of 1 approvals obtained.

**Comment by Abhishek-CDC** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java` line 1693 at r3](https://reviewable.io/reviews/harness/harness-core/35893#-N8Nt5RDj1zkMcSdO1US-r3-1693:-N8gYPcm4gLbeUYHjDfL:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5ed360ba1ca3bf15378c1dc8d7d848919884ba7b/120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java#L1693)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

In general, as a good coding practise, we should avoid returning null from a method (https://www.codebyamir.com/blog/stop-returning-null-in-java). You can return Optional<String> in this case or an empty string
</blockquote></details>

Done.

**Comment by Abhishek-CDC** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java` line 1701 at r3](https://reviewable.io/reviews/harness/harness-core/35893#-N8Nt5RDj1zkMcSdO1US-r3-1701:-N8gYRnJCA9GcCgtdNjW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5ed360ba1ca3bf15378c1dc8d7d848919884ba7b/120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java#L1701)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

same comment for returning null as above
</blockquote></details>

Done.

**Comment by Abhishek-CDC** [comment]
*[`126-instance/src/main/java/io/harness/service/deploymentevent/DeploymentEventListener.java` line 187 at r3](https://reviewable.io/reviews/harness/harness-core/35893#-N8Nt5RDj1zkMcSdO1UX-r3-187:-N8gYUDcAUSqWptd51qA:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5ed360ba1ca3bf15378c1dc8d7d848919884ba7b/126-instance/src/main/java/io/harness/service/deploymentevent/DeploymentEventListener.java#L187)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

can we call this field displayName itself ?
</blockquote></details>

Done.

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 6 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35893)* status: 1 file(s) to review. 0 of 1 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 7 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35893)* status: :shipit: complete! ✓

**Comment by Abhishek-CDC** [comment]
trigger ti3

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35893)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35893)* status: :shipit: complete! ✓

**Pull request merged**